runtime error: slice bounds out of range?

I'm spinning up a test instance of Consul, but I keep getting this `slice bounds out of range` error. This is an Ubuntu 14.04 LTS Vagrant box. I've got golang 1.4.2 installed on here.

I ran the following:

1. `git clone https://github.com/hashicorp/envconsul.git`
2. `cd envconsul`
3. `mkdir $HOME/go`
4. `export GOPATH=$HOME/go`
5. `make`
6. ` ./consul agent -server -bootstrap-expect 1 -data-dir /tmp/consul &`
7. `curl -X PUT -d 'test' http://localhost:8500/v1/kv/web/key1` (returns true)

I then hit envconsul:

```
vagrant@vagrant-ubuntu-trusty-64:~$ ./envconsul/bin/envconsul -consul=127.0.0.1:8500 -prefix web
panic: runtime error: slice bounds out of range
```

Full dump is [here](https://gist.github.com/greggilbert/e0893a100cbcf1b22afd). I'm not sure what I'm doing wrong; this all seems correct per the documentation. Any ideas? Thanks!

--- a/cli.go
+++ b/cli.go
@@ -70,11 +70,6 @@ func (cli *CLI) Run(args []string) int {
 		return cli.handleError(err, ExitCodeParseFlagsError)
 	}
 
-	// Return an error if no command was given
-	if len(command) == 0 {
-		return cli.handleError(ErrMissingCommand, ExitCodeUsageError)
-	}
-
 	// If a path was given, load the config from file
 	if config.Path != "" {
 		newConfig, err := ConfigFromPath(config.Path)
@@ -96,6 +91,11 @@ func (cli *CLI) Run(args []string) int {
 		Writer:         cli.errStream,
 	}); err != nil {
 		return cli.handleError(err, ExitCodeLoggingError)
+	}
+
+	// Return an error if no command was given
+	if len(command) == 0 && !version {
+		return cli.handleError(ErrMissingCommand, ExitCodeUsageError)
 	}
 
 	// Print version information for debugging
@@ -331,7 +331,7 @@ func (cli *CLI) parseFlags(args []string) (*Config, []string, bool, bool, error)
 // handleError outputs the given error's Error() to the errStream and returns
 // the given exit status.
 func (cli *CLI) handleError(err error, status int) int {
-	cli.errStream.Write([]byte(err.Error() + "\n"))
+	log.Printf("[ERR] %s", err.Error())
 	return status
 }
 

--- a/cli.go
+++ b/cli.go
@@ -30,6 +30,12 @@ const (
 	ExitCodeParseFlagsError
 	ExitCodeRunnerError
 	ExitCodeConfigError
+	ExitCodeUsageError
+)
+
+var (
+	// Error returned when no command is specified.
+	ErrMissingCommand = fmt.Errorf("No command given")
 )
 
 /// ------------------------- ///
@@ -62,6 +68,11 @@ func (cli *CLI) Run(args []string) int {
 	config, command, once, version, err := cli.parseFlags(args[1:])
 	if err != nil {
 		return cli.handleError(err, ExitCodeParseFlagsError)
+	}
+
+	// Return an error if no command was given
+	if len(command) == 0 {
+		return cli.handleError(ErrMissingCommand, ExitCodeUsageError)
 	}
 
 	// If a path was given, load the config from file
@@ -320,12 +331,12 @@ func (cli *CLI) parseFlags(args []string) (*Config, []string, bool, bool, error)
 // handleError outputs the given error's Error() to the errStream and returns
 // the given exit status.
 func (cli *CLI) handleError(err error, status int) int {
-	log.Printf("[ERR] %s", err.Error())
+	cli.errStream.Write([]byte(err.Error() + "\n"))
 	return status
 }
 
 const usage = `
-Usage: %s [options]
+Usage: %s [options] <command>
 
   Watches values from Consul's K/V store and sets environment variables when
   Consul values are changed.

--- a/cli_test.go
+++ b/cli_test.go
@@ -1,6 +1,7 @@
 package main
 
 import (
+	"bytes"
 	"io/ioutil"
 	"reflect"
 	"strings"
@@ -492,5 +493,21 @@ func TestParseFlags_errors(t *testing.T) {
 
 	if err == nil {
 		t.Fatal("expected error, but nothing was returned")
+	}
+}
+
+func TestRun_errors(t *testing.T) {
+	buf := new(bytes.Buffer)
+
+	// Returns the right exit code if no command is given
+	cli := NewCLI(ioutil.Discard, buf)
+	if code := cli.Run([]string{"envconsul"}); code != ExitCodeUsageError {
+		t.Fatalf("expected %d, got: %d", ExitCodeUsageError, code)
+	}
+
+	// Output reflects the returned error
+	out := buf.String()
+	if !strings.Contains(out, ErrMissingCommand.Error()) {
+		t.Fatalf("expected to find %q, got: %q", ErrMissingCommand.Error(), out)
 	}
 }

--- a/cli_test.go
+++ b/cli_test.go
@@ -510,4 +510,9 @@ func TestRun_errors(t *testing.T) {
 	if !strings.Contains(out, ErrMissingCommand.Error()) {
 		t.Fatalf("expected to find %q, got: %q", ErrMissingCommand.Error(), out)
 	}
+
+	// Version flag still works
+	if code := cli.Run([]string{"envconsul", "-v"}); code != 0 {
+		t.Fatalf("version command exited %d", code)
+	}
 }

